azure-commons-core/src/main/java/com/microsoft/jenkins/azurecommons/core/credentials/ImdsTokenCredentials.java

```diff
@@ -41,10 +41,11 @@ protected static String requestIMDSEndpoint(final String resource) throws IOExce
                 .addInterceptor(loggingInterceptor)
                 .build();
 
-        String parameters = "api-version=2018-02-01&resource=https://management.azure.com/";
+        String parameters = "https://management.azure.com/";
         Request request = new Request.Builder()
                 .addHeader("Metadata", "true")
                 .url("http://169.254.169.254/metadata/identity/oauth2/token?"
+                        + "api-version=2018-02-01&resource="
```

Comment:
URLEncode?

Reply:
Encoding the whole parameter string here won't work. Just encode the URL parameter.